lib/features/thread/data/network/thread_api.dart

```diff
@@ -226,57 +233,98 @@ class ThreadAPI {
+        ..usings(usedCapabilities))
       .build()
       .execute();
 
     final resultChanges = result.parse<ChangesEmailResponse>(
       changesEmailInvocation.methodCallId,
       ChangesEmailResponse.deserialize);
 
-    final resultUpdated = result.parse<GetEmailResponse>(
-      getEmailUpdatedInvocation.methodCallId,
-      GetEmailResponse.deserialize);
-
-    final resultCreated = result.parse<GetEmailResponse>(
-      getEmailCreatedInvocation.methodCallId,
-      GetEmailResponse.deserialize);
-
-    final listMailboxIdDestroyed = resultChanges?.destroyed.map((id) => EmailId(id)).toList() ?? <EmailId>[];
+    List<EmailId> destroyedEmailIds = resultChanges
+      ?.destroyed
+      .toEmailIds()
+      .toList() ?? [];
+    State? newStateChanges = resultChanges?.newState;
+    bool hasMoreChanges = resultChanges?.hasMoreChanges ?? false;
+    List<Email>? updatedEmail;
+    List<Email>? createdEmail;
+    State? newStateEmail;
+
+    if (getEmailUpdatedInvocation != null) {
+      final emailResponseUpdated = result.parse<GetEmailResponse>(
+        getEmailUpdatedInvocation.methodCallId,
+        GetEmailResponse.deserialize,
+      );
+      updatedEmail = emailResponseUpdated?.list;
+      newStateEmail = emailResponseUpdated?.state;
+      final notFoundIdsUpdated = emailResponseUpdated?.notFound?.toEmailIds().toList() ?? [];
+      log('ThreadAPI::getChanges:notFoundIdsUpdated = $notFoundIdsUpdated');
+      destroyedEmailIds.addAll(notFoundIdsUpdated);
+    }
 
+    if (getEmailCreatedInvocation != null) {
+      final emailResponseCreated = result.parse<GetEmailResponse>(
+        getEmailCreatedInvocation.methodCallId,
+        GetEmailResponse.deserialize,
+      );
+      createdEmail = emailResponseCreated?.list;
+      newStateEmail = emailResponseCreated?.state;
+      final notFoundIdsCreated = emailResponseCreated?.notFound?.toEmailIds().toList() ?? [];
+      log('ThreadAPI::getChanges:notFoundIdsCreated = $notFoundIdsCreated');
+      destroyedEmailIds.addAll(notFoundIdsCreated);
+    }
+    log('ThreadAPI::getChanges:newStateChanges = $newStateChanges | newStateEmail = $newStateEmail | hasMoreChanges = $hasMoreChanges');
+    log('ThreadAPI::getChanges:updatedEmailSize = ${updatedEmail?.length} | createdEmailSize = ${createdEmail?.length}');
+    log('ThreadAPI::getChanges:destroyedEmailIds = $destroyedEmailIds');
     return EmailChangeResponse(
-      updated: resultUpdated?.list,
-      created: resultCreated?.list,
-      destroyed: listMailboxIdDestroyed,
-      newStateEmail: resultUpdated?.state,
-      newStateChanges: resultChanges?.newState,
-      hasMoreChanges: resultChanges?.hasMoreChanges ?? false,
-      updatedProperties: propertiesUpdated);
+      updated: updatedEmail,
+      created: createdEmail,
+      destroyed: destroyedEmailIds,
+      newStateEmail: newStateEmail,
+      newStateChanges: newStateChanges,
+      hasMoreChanges: hasMoreChanges,
+      updatedProperties: propertiesUpdated,
```

Comment:
why not add `createdProperties` here? IMO, we should make sure the loop call to this never put it null.

Reply:
We don't need it